Doppler.PushContact/Services/Messages/MessageSender.cs

```diff
@@ -0,0 +1,65 @@
+using Doppler.PushContact.Services.Messages.ExternalContracts;
+using Flurl;
+using Flurl.Http;
+using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Doppler.PushContact.Services.Messages
+{
+    public class MessageSender : IMessageSender
+    {
+        private readonly MessageSenderSettings _messageSenderSettings;
+        private readonly IPushApiTokenGetter _pushApiTokenGetter;
+
+        public MessageSender(IOptions<MessageSenderSettings> messageSenderSettings, IPushApiTokenGetter pushApiTokenGetter)
+        {
+            _messageSenderSettings = messageSenderSettings.Value;
+            _pushApiTokenGetter = pushApiTokenGetter;
+        }
+
+        public async Task<IEnumerable<SendMessageTargetResult>> SendAsync(string title, string body, IEnumerable<string> targetDeviceTokens)
+        {
+            if (string.IsNullOrEmpty(title))
+            {
+                throw new ArgumentException($"'{nameof(title)}' cannot be null or empty.", nameof(title));
+            }
+
+            if (string.IsNullOrEmpty(body))
+            {
+                throw new ArgumentException($"'{nameof(body)}' cannot be null or empty.", nameof(body));
+            }
+
+            if (targetDeviceTokens == null || !targetDeviceTokens.Any())
+            {
+                throw new ArgumentException($"'{nameof(targetDeviceTokens)}' cannot be null or empty.", nameof(targetDeviceTokens));
+            }
+
+            var pushApiToken = await _pushApiTokenGetter.GetTokenAsync();
+
+            var responseBody = await _messageSenderSettings.PushApiUrl
+                    .AppendPathSegment("message")
+                    .WithOAuthBearerToken(pushApiToken)
+                    .PostJsonAsync(new
+                    {
+                        notificationTitle = title,
+                        notificationBody = body,
+                        tokens = targetDeviceTokens
+                    })
+                    .ReceiveJson<SendMessageResponse>();
+
+            return responseBody.Responses.Select(x =>
+                {
+                    return new SendMessageTargetResult
+                    {
+                        TargetDeviceToken = x.DeviceToken,
+                        IsSuccess = x.IsSuccess,
+                        IsValidTargetDeviceToken = !x.IsSuccess && _messageSenderSettings.FatalMessagingErrorCodes.Any(y => y == x.Exception.MessagingErrorCode),
+                        NotSuccessErrorDetails = !x.IsSuccess ? $"{nameof(x.Exception.MessagingErrorCode)} {x.Exception.MessagingErrorCode} - {nameof(x.Exception.Message)} {x.Exception.Message}" : null
+                    };
+                });
```

Comment:
Creo que acá también

```suggestion
            return responseBody.Responses.Select(x =>
            {
                return new SendMessageTargetResult
                {
                    TargetDeviceToken = x.DeviceToken,
                    IsSuccess = x.IsSuccess,
                    IsValidTargetDeviceToken = !x.IsSuccess && _messageSenderSettings.FatalMessagingErrorCodes.Any(y => y == x.Exception.MessagingErrorCode),
                    NotSuccessErrorDetails = !x.IsSuccess ? $"{nameof(x.Exception.MessagingErrorCode)} {x.Exception.MessagingErrorCode} - {nameof(x.Exception.Message)} {x.Exception.Message}" : null
                };
            });
```